src/Api/OperationTypeDeprecationHelper.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ApiPlatform\Core\Api;
+
+/**
+ * Class OperationTypeDeprecationHelper
+ * Before API Platform 2.1, the operation type was one of:
+ * - "collection" (true)
+ * - "item" (false).
+ *
+ * Because we introduced a third type in API Platform 2.1, we're using a string with OperationType constants:
+ * - OperationType::ITEM
+ * - OperationType::COLLECTION
+ * - OperationType::SUBCOLLECTION
+ */
+final class OperationTypeDeprecationHelper
```

Comment:
Not really useful to have a trait for this, static methods are fine enough IMO but I can change this.